cmake/Abseil.cmake

```diff
@@ -0,0 +1,29 @@
+set(ABSL_ENABLE_INSTALL ON CACHE BOOL "" FORCE)
+if (NOT (DEFINED ABSEIL_PATH))
+  message("ABSEIL_PATH not provided. Downloading.")
+  set(ABSEIL_PREFIX "${benchmark_BINARY_DIR}/third_party/abseil")
+  configure_file(${benchmark_SOURCE_DIR}/cmake/Abseil.cmake.in ${ABSEIL_PREFIX}/CMakeLists.txt @ONLY)
+  execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
```

Comment:
This doesn't pass through CMake options to the Abseil build and instead relies on defaults. For example, CMake will use `/usr/bin/cc` and `/usr/bin/c++` as the default compiler on UNIX-like systems but that may be undesirable. In our case, it's going to fail because those binaries don't exists on our bots at all. Rather, we always specify the compiler manually via `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER`. The same applies to all other tools. Ideally we would pass all of these variables from the benchmark build to the abseil build.